components/match2/commons/match_group_display_singlematch.lua

```diff
@@ -0,0 +1,108 @@
+---
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:MatchGroup/Display/SingleMatch
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local DisplayHelper = require('Module:MatchGroup/Display/Helper')
+local DisplayUtil = require('Module:DisplayUtil')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+local TypeUtil = require('Module:TypeUtil')
+
+local MatchGroupUtil = Lua.import('Module:MatchGroup/Util', {requireDevIfEnabled = true})
+
+local SingleMatchDisplay = {propTypes = {}, types = {}}
+
+SingleMatchDisplay.configFromArgs = function(args)
+	return {
+		width = tonumber(string.gsub(args.width or '', 'px', ''), nil),
+	}
+end
+
+SingleMatchDisplay.types.SingleMatchConfig = TypeUtil.struct({
+	MatchSummaryContainer = 'function',
+	width = 'number',
+})
+SingleMatchDisplay.types.SingleMatchConfigOptions = TypeUtil.struct(
+	Table.mapValues(SingleMatchDisplay.types.SingleMatchConfig.struct, TypeUtil.optional)
+)
+
+SingleMatchDisplay.propTypes.SingleMatchContainer = {
+	bracketId = 'string',
+	config = TypeUtil.optional(SingleMatchDisplay.types.SingleMatchConfigOptions),
+}
+
+--[[
+Display component for a singleMatch. The singleMatch is specified by ID.
+The component fetches the match data from LPDB or page variables.
+]]
+function SingleMatchDisplay.SingleMatchContainer(props)
+	DisplayUtil.assertPropTypes(props, SingleMatchDisplay.propTypes.SingleMatchContainer)
+	return SingleMatchDisplay.SingleMatch({
+		config = props.config,
+		match = props.match,
+	})
+end
```

Comment:
By convention display components named `FooContainer` render a 'foo' by id rather than a foo struct. The id is used by the container component to retrieve the 'foo' (say from lpdb), then the foo is passed to the non-container display component.

Here's how `SingleMatchContainer` would look if it takes an id instead of match object: (I put the prop.match nil check originally in `SingleMatchDisplay.SingleMatch` here)

```


SingleMatchDisplay.propTypes.SingleMatchContainer = {
	bracketId = 'string',
	matchId = 'string',
	config = TypeUtil.optional(SingleMatchDisplay.types.SingleMatchConfigOptions),
}

--[[
Display component for a single match. The single match is specified by ID.
The component fetches the match data from LPDB or page variables.
]]
function SingleMatchDisplay.SingleMatchContainer(props)
	DisplayUtil.assertPropTypes(props, SingleMatchDisplay.propTypes.SingleMatchContainer)

	local matchGroup = MatchGroupUtil.fetchMatchGroup(props.bracketId)
	local match = matchGroup.matchesById[props.matchId]
	return match
		and SingleMatchDisplay.SingleMatch({
			config = props.config,
			match = match,
		})
		or ''
end


```

Reply:
Implemented as suggested